### PR TITLE
[WIP] temperature calibration

### PR DESCRIPTION
**Pull request by dagar**
@HesselM this is your PR #1943 rebased on master.

**todo**
- [ ] mpu6k compensation Do NOT store it in a separate variable, but just store everything right into the normal rad/s and m/s2 fields https://github.com/PX4/Firmware/compare/master...dagar:tempcal_rebase?expand=1#diff-afefce4f4a086c7a7d23e52028f9ae1eR306
- [ ] only leave the bare minimum required for testing. We don't want to blow up log rates by a factor of 2 whenever feasible.
- [ ] calibration workflow
- [ ] testing

### CODE REVIEW

**Comment by LorenzMeier** [comment]
Ok, the diff with the additional log message does look less horrible than I thought. Lets leave it in for the time being and gather some user-logs with it.

File: src/modules/sdlog2/sdlog2_messages.h [hunk 1]
```diff
@@ -519,10 +520,11 @@ struct log_PARM_s {
 static const struct log_format_s log_formats[] = {
 	/* business-level messages, ID < 0x80 */
 	LOG_FORMAT(ATT, "fffffffffffff",	"qw,qx,qy,qz,Roll,Pitch,Yaw,RollRate,PitchRate,YawRate,GX,GY,GZ"),
-	LOG_FORMAT(ATSP, "ffffffff",		"RollSP,PitchSP,YawSP,ThrustSP,qw,qx,qy,qz"),
-	LOG_FORMAT_S(IMU, IMU, "ffffffffffff",		"AccX,AccY,AccZ,GyroX,GyroY,GyroZ,MagX,MagY,MagZ,tA,tG,tM"),
+	LOG_FORMAT(ATSP, "ffff",		"RollSP,PitchSP,YawSP,ThrustSP"),
```
Comment (LorenzMeier) [inline src/modules/sdlog2/sdlog2_messages.h]: Merge fail: The ATSP message must not change

Comment (dagar) [inline src/modules/sdlog2/sdlog2_messages.h]: fixed and rebased

**Comment by LorenzMeier** [comment]
ok, then lets get this tested! Once we confirm it doesn't break anything we can move ahead to more testing with a wider circle.

**Comment by dagar** [comment]
Weird build failure that I'm not seeing locally. Rebasing to restart the build.

**Comment by dagar** [comment]
Anyone know why this is only happening on px4fmu-v4?

    arm-none-eabi/bin/ld: address 0x8100d70 of src/firmware/nuttx/firmware_nuttx section `__param' is not within region `flash'
    arm-none-eabi/bin/ld: src/firmware/nuttx/firmware_nuttx section `.ARM.exidx' will not fit in region `flash'
    arm-none-eabi/bin/ld: address 0x8100d70 of src/firmware/nuttx/firmware_nuttx section `__param' is not within region `flash'
    arm-none-eabi/bin/ld: address 0x8100d70 of src/firmware/nuttx/firmware_nuttx section `__param' is not within region `flash'
    arm-none-eabi/bin/ld: region `flash' overflowed by 6340 bytes

**Comment by davids5** [comment]
Try 
git submodule sync
make clean
make px4fmu-v4_default

Does it still happen?

On December 17, 2015 2:53:09 PM HST, Daniel Agar <notifications@github.com> wrote:
>Anyone know why this is only happening on px4fmu-v4?
>
>arm-none-eabi/bin/ld: address 0x8100d70 of
>src/firmware/nuttx/firmware_nuttx section `__param' is not within
>region `flash'
>arm-none-eabi/bin/ld: src/firmware/nuttx/firmware_nuttx section
>`.ARM.exidx' will not fit in region `flash'
>arm-none-eabi/bin/ld: address 0x8100d70 of
>src/firmware/nuttx/firmware_nuttx section `__param' is not within
>region `flash'
>arm-none-eabi/bin/ld: address 0x8100d70 of
>src/firmware/nuttx/firmware_nuttx section `__param' is not within
>region `flash'
>    arm-none-eabi/bin/ld: region `flash' overflowed by 6340 bytes
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/PX4/Firmware/pull/3336#issuecomment-165627042

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by dagar** [comment]
It's happening on travis-ci https://travis-ci.org/PX4/Firmware/jobs/97573607#L1597 both Linux and OSX. Weirdly I can't reproduce it locally on Ubuntu 14.04 with the same gcc 4.8.4.

**Comment by LorenzMeier** [comment]
@dagar If you rebase it on current master it will pass now. I missed to enable the full 2M flash on Pixracer.
https://github.com/PX4/Firmware/commit/699c5f2f6d5c3b47a6a386b9f089924f032d7c6a

Let me know if you have flight tested this and tested that the logs are good. Please also share the post-flight performance counters file so we can check load.

**Comment by dagar** [comment]
The rebase fixed the build failure. I'll get a test flight in tomorrow.

**Comment by dagar** [comment]
I did a few test flights that went fine, but the IMUT logging isn't right. Looking.

http://logs.uaventure.com/view/CtD57unvj6cvYSHd9MXWME
preflight - https://gist.github.com/dagar/fb91feb7d9f02a65eb5d
postflight - https://gist.github.com/dagar/8c0090c48e7cec1d36f9

**Comment by dagar** [comment]
Found the problem - LOG_IMUT_MSG and LOG_VTOL_MSG are both defined as 43, which makes sense given the timeframe of the merge. Fixed with a comment to avoid future conflict and rebasing.

**Comment by LorenzMeier** [comment]
@dagar how is the testing state looking here? I would be eager to get it in, at least as a test case to gather data.

**Comment by dagar** [comment]
I flew again after my last comment, but I'm still getting logs with invalid headers.
`Exception: Invalid header at 23793 (0x5CF1): A0 8A, must be A3 95`

It might be better to handle this logging with a new msg type anyway.
https://github.com/dagar/Firmware/blob/tempcal_rebase/src/modules/sdlog2/sdlog2.c#L1474

**Pull request closed without merging**